lib/sqlalchemy/dialects/sybase/base.py

```diff
@@ -515,22 +515,20 @@ class SybaseSQLCompiler(compiler.SQLCompiler):
     )
 
     def get_select_precolumns(self, select, **kw):
-        s = select._distinct and "DISTINCT " or ""
-
-        if select._simple_int_limit and not select._offset:
-            kw["literal_execute"] = True
-            s += "TOP %s " % self.process(select._limit_clause, **kw)
-
-        if select._offset:
-            raise NotImplementedError("Sybase ASE does not support OFFSET")
-        return s
+        return select._distinct and "DISTINCT " or ""
```

Comment:
This could probably be removed entirely, since the default version would work
https://github.com/sqlalchemy/sqlalchemy/blob/5311a8531c621760ced0b3d0a5454e8c22871d66/lib/sqlalchemy/sql/compiler.py#L2866-L2880

Reply:
Good call. Just removed it :+1: